operator/pkg/metrics/monitoring.go

```diff
@@ -0,0 +1,237 @@
+
+var (
+	// CRFetchErrorReasonLabel describes the reason/HTTP code
+	// for failing to fetch CR.
+	CRFetchErrorReasonLabel = monitoring.MustCreateLabel("reason")
+
+	// CRNamespacedNameLabel describes the name with namespace.
+	CRNamespacedNameLabel = monitoring.MustCreateLabel("name")
+)
+
+var (
+	// ComponentNameLabel represents istio component name - like
+	// core, pilot, istio-cni etc.
+	ComponentNameLabel = monitoring.MustCreateLabel("component")
+
+	// ResourceKindLabel indicates the kind of resource owned
+	// or created or updated or deleted or pruned by operator.
+	ResourceKindLabel = monitoring.MustCreateLabel("kind")
+
+	// CRRevisionLabel indicates the revision in the CR.
+	CRRevisionLabel = monitoring.MustCreateLabel("revision")
+)
+
+// MergeErrorType describes the class of errors that could
+// occur while merging profile, user supplied YAML, values
+// overridden by --set and so on.
+type MergeErrorType string
+
+const (
+	// CannotFetchProfileError occurs when profile cannot be found.
+	CannotFetchProfileError MergeErrorType = "cannot_fetch_profile"
+
+	// OverlayError overlaying YAMLs to combine profile, user
+	// defined settings in CR, Hub-tag etc fails.
+	OverlayError MergeErrorType = "overlay"
+
+	// IOPFormatError occurs when supplied CR cannot be marshaled
+	// or unmarshaled to/from YAML.
+	IOPFormatError MergeErrorType = "iop_format"
+
+	// TranslateValuesError occurs when translating from legacy API fails.
+	TranslateValuesError MergeErrorType = "translate_values"
+
+	// InternalYAMLParseError occurs when spec section in merged CR
+	// cannot be accessed for some reason (either missing or multiple).
+	InternalYAMLParseError MergeErrorType = "internal_yaml_parse"
+)
+
+// RenderErrorType describes the class of errors that could
+// occur while rendering Kubernetes manifest from given CR.
+type RenderErrorType string
+
+const (
+	RenderNotStartedError RenderErrorType = "render_not_started"
+
+	// HelmTranslateIOPToValuesError describes render error where renderer for
+	// a component cannot create values.yaml tree from given CR.
+	HelmTranslateIOPToValuesError RenderErrorType = "helm_translate_iop_to_values"
+
+	// HelmChartRenderError describes error where Helm charts cannot be rendered
+	// for the generated values.yaml tree.
+	HelmChartRenderError RenderErrorType = "helm_chare_render"
+
+	// K8SSettingsOverlayError describes the K8s overlay error after
+	// rendering Helm charts successfully.
+	K8SSettingsOverlayError RenderErrorType = "k8s_settings_overlay"
+
+	// K8SManifestPatchError describes errors while patching generated manifest.
+	K8SManifestPatchError RenderErrorType = "k8s_manifest_patch"
+)
+
+var (
+	// Version is the version of the operator binary running currently.
+	// This is required for fleet level metrics although it is available from
+	// ControlZ (more precisely versionz endpoint).
+	Version = monitoring.NewGauge(
+		"istio_install_operator_version",
```

Comment:
nit: if all of the these metrics are intended to begin with `istio_install_operator`, you could consider using the prom libraries when configuring the exporters to wrap the `Registerer` with a prefix of `istio_install_operator`.  Then if you ever wanted to change it, it would be a one-line change.

Reply:
Oh! Yes. Fixed it. That's `Namespace` :)